From facc351c59a49bf1c08f13ce42dfc047d45e23cd Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Wed, 6 Aug 2014 03:46:39 -0400
Subject: [pci] Remove pci_fixup_parent_subordinate_busnr()

Message-id: <20140806034639.1438.85514.stgit@gir.stowe>
Patchwork-id: 87433
O-Subject: [RHEL7.1 PATCH 24/69] PCI: Remove pci_fixup_parent_subordinate_busnr()
Bugzilla: 1110896
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110896
Upstream Status: f5fb40700fc9a52944fbe07148c858a5025908b1

commit f5fb40700fc9a52944fbe07148c858a5025908b1
Author: Andreas Noever <andreas.noever@gmail.com>
Date:   Thu Jan 23 21:59:26 2014 +0100

    PCI: Remove pci_fixup_parent_subordinate_busnr()

    The function has no effect.

    If pcibios_assign_all_busses() is not set then the function does nothing.

    If it is set then in pci_scan_bridge we are always in the branch where
    we assign the bus numbers ourselves and the subordinate values of all
    parent busses will be set to 0xff since that is what they inherited from
    their parent bus and ultimately from the root bus.

    Signed-off-by: Andreas Noever <andreas.noever@gmail.com>
    Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/pci/probe.c b/drivers/pci/probe.c
index 0dac968..ba0bd70 100644
--- a/drivers/pci/probe.c
+++ b/drivers/pci/probe.c
@@ -774,22 +774,6 @@ struct pci_bus *__ref pci_add_new_bus(struct pci_bus *parent, struct pci_dev *de
  return child;
 }
 
-static void pci_fixup_parent_subordinate_busnr(struct pci_bus *child, int max)
-{
- struct pci_bus *parent = child->parent;
-
- /* Attempts to fix that up are really dangerous unless
-    we're going to re-assign all bus numbers. */
- if (!pcibios_assign_all_busses())
-  return;
-
- while (parent->parent && parent->busn_res.end < max) {
-  parent->busn_res.end = max;
-  pci_write_config_byte(parent->self, PCI_SUBORDINATE_BUS, max);
-  parent = parent->parent;
- }
-}
-
 /*
  * If it's a bridge, configure it and scan the bus behind it.
  * For CardBus bridges, we don't scan behind as the devices will
@@ -922,20 +906,7 @@ int pci_scan_bridge(struct pci_bus *bus, struct pci_dev *dev, int max, int pass)
 
   if (!is_cardbus) {
    child->bridge_ctl = bctl;
-   /*
-    * Adjust subordinate busnr in parent buses.
-    * We do this before scanning for children because
-    * some devices may not be detected if the bios
-    * was lazy.
-    */
-   pci_fixup_parent_subordinate_busnr(child, max);
-   /* Now we can scan all subordinate buses... */
    max = pci_scan_child_bus(child);
-   /*
-    * now fix it up again since we have found
-    * the real value of max.
-    */
-   pci_fixup_parent_subordinate_busnr(child, max);
   } else {
    /*
     * For CardBus bridges, we leave 4 bus numbers
@@ -966,7 +937,6 @@ int pci_scan_bridge(struct pci_bus *bus, struct pci_dev *dev, int max, int pass)
     }
    }
    max += i;
-   pci_fixup_parent_subordinate_busnr(child, max);
   }
   /*
    * Set the subordinate bus number to its real value.
-- 
1.7.1